### ISSUE TITLE
Error compiling Bitcoin Core on Centos 7.5 (with default gcc 4.8)

### ISSUE THREAD

**Issue by martinbehrens** [bitcoin/bitcoin#15172]
***Description***

Compiling bitcoind (cf0c67b62c2037dc9e70ea84ffee3b205a9b1bef, master branch) fails with:

```
/usr/include/c++/4.8.2/bits/stl_list.h:1149:9: note:   template argument deduction/substitution failed:
rpc/server.cpp:54:122: note:   cannot convert ‘g_rpc_server_info.RPCServerInfo::active_commands.std::list<_Tp, _Alloc>::cend<RPCCommandExecutionInfo, std::allocator<RPCCommandExecutionInfo> >()’ (type ‘std::list<RPCCommandExecutionInfo>::const_iterator {aka std::_List_const_iterator<RPCCommandExecutionInfo>}’) to type ‘std::list<RPCCommandExecutionInfo>::iterator {aka std::_List_iterator<RPCCommandExecutionInfo>}’
         it = g_rpc_server_info.active_commands.insert(g_rpc_server_info.active_commands.cend(), {method, GetTimeMicros()});
```

***Steps to replicate***

Replicable on two different VM's:

(- git checkout master)
- git pull
- make clean && ./autogen.sh && ./configure --disable-wallet --without-gui --with-zmq --enable-zmq && make

***Expected behavior***

Compilation should be successful.

***Current behavior***

Compilation was not successful.

***Bitcoin Core Version***

```
[user@localhost bitcoin]$ git branch
* master
[user@localhost bitcoin]$ git log
commit cf0c67b62c2037dc9e70ea84ffee3b205a9b1bef

```
***Environment Info***

```
[user@localhost bitcoin]$ cat /etc/centos-release
CentOS Linux release 7.5.1804 (Core)
```

```
[user@localhost bitcoin]$ cat /proc/cpuinfo
processor	: 0
vendor_id	: GenuineIntel
cpu family	: 6
model		: 42
model name	: Intel Xeon E312xx (Sandy Bridge)
stepping	: 1
microcode	: 0x1
cpu MHz		: 3092.972
cache size	: 16384 KB
physical id	: 0
siblings	: 1
core id		: 0
cpu cores	: 1
apicid		: 0
initial apicid	: 0
fpu		: yes
fpu_exception	: yes
cpuid level	: 13
wp		: yes
flags		: fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ss syscall nx rdtscp lm constant_tsc rep_good nopl xtopology eagerfpu pni pclmulqdq ssse3 cx16 pcid sse4_1 sse4_2 x2apic popcnt tsc_deadline_timer aes xsave avx hypervisor lahf_lm xsaveopt arat
bogomips	: 6185.94
clflush size	: 64
cache_alignment	: 64
address sizes	: 40 bits physical, 48 bits virtual
power management:
```

```
[user@localhost bitcoin]$ gcc -v
Using built-in specs.
COLLECT_GCC=gcc
COLLECT_LTO_WRAPPER=/usr/libexec/gcc/x86_64-redhat-linux/4.8.5/lto-wrapper
Target: x86_64-redhat-linux
Configured with: ../configure --prefix=/usr --mandir=/usr/share/man --infodir=/usr/share/info --with-bugurl=http://bugzilla.redhat.com/bugzilla --enable-bootstrap --enable-shared --enable-threads=posix --enable-checking=release --with-system-zlib --enable-__cxa_atexit --disable-libunwind-exceptions --enable-gnu-unique-object --enable-linker-build-id --with-linker-hash-style=gnu --enable-languages=c,c++,objc,obj-c++,java,fortran,ada,go,lto --enable-plugin --enable-initfini-array --disable-libgcj --with-isl=/builddir/build/BUILD/gcc-4.8.5-20150702/obj-x86_64-redhat-linux/isl-install --with-cloog=/builddir/build/BUILD/gcc-4.8.5-20150702/obj-x86_64-redhat-linux/cloog-install --enable-gnu-indirect-function --with-tune=generic --with-arch_32=x86-64 --build=x86_64-redhat-linux
Thread model: posix
gcc version 4.8.5 20150623 (Red Hat 4.8.5-28) (GCC)
```

**Comment by martinbehrens** [comment]
Note: Still reproducible at a5daf70ffb80c14ddf9f35c638d9e8087d8732e9

**Comment by MarcoFalke** [comment]
This could be the gcc version, unfortunately we don't test that Bitcoin Core still compiles with the minimum required (4.8)

**Comment by martinbehrens** [comment]
> This could be the gcc version, unfortunately we don't test that Bitcoin Core still compiles with the minimum required (4.8)

Confirmed.

After upgrading gcc with ...
```
yum install centos-release-scl
yum install devtoolset-7-gcc*
scl enable devtoolset-7 bash
```
... to v7.3.1 it does compile successfully.

**Comment by MarcoFalke** [comment]
I proposed a fix in  #15248
